addons/escoria-core/game/scenes/dialogs/esc_dialog_player.gd

```diff
@@ -67,41 +37,93 @@ func _add_states_to_machine() -> void:
 # - type: UI to use for the dialog
 # - text: Text to say
 func say(character: String, type: String, text: String) -> void:
-	states_map["say"].initialize(character, type, text)
-	_change_state("say")
+	if type == "":
+		type = ESCProjectSettingsManager.get_setting(
+			ESCProjectSettingsManager.DEFAULT_DIALOG_TYPE
+		)
 
+	_determine_say_dialog_manager(type)
 
-# Called when a dialogue line is to be sped up.
-func speedup() -> void:
-	_change_state("say_fast")
+	if is_a_parent_of(_say_dialog_manager):
+		remove_child(_say_dialog_manager)
 
+	add_child(_say_dialog_manager)
```

Comment:
I am always scared when I see an `add_child()` clause somewhere. This `say()` method is supposed to be called multiple times, right? So it is quite possible that the `_say_dialog_manager` node is added as child each time the `say()` method is called.